### DISCUSSION TITLE
Do we need to assume some parametric model to utilize lifelines?

### DISCUSSION THREAD

**Discussion by ogencoglu** [CamDavidsonPilon/lifelines#1628] [Q&A] [answered]
Do we need to assume some parametric model to utilize lifelines? Also can one have an arbitrary code as the underlying assumed model (if-else, non-linear and possibly non-convex stuff - basically a black box)?

**Comment by pzivich** [comment]
No, there is support for both semiparametric and nonparametric models. For example, the Cox proportional hazards model is semiparametric, since no model is specified for the baseline hazard. Similarly, the Kaplan-Meier and Nelson-Aalen estimators are nonparametric estimators. Finally, you can also pair `lifelines` with other estimators. For example, you can pair the Kaplan-Meier with inverse probability weighting for causal effect estimation.

However, I think your question is trying to see if there is an option for machine learning. `lifelines` does support some penalized models (which you could manually pair with splines to get a GAM). To my knowledge, `lifelines` does not easily integrate support for arbitrary code for the model but someone might know better

**Reply by ogencoglu** [reply]
Thanks for the answer. Indeed that was my question: can I use a machine learning model or arbitrary code.

**Reply by CamDavidsonPilon** [reply, marked as answer]
A lot of survival analysis in lifelines involves a likelihood equation, and this is provided by starting with a parametric model. The likelihood equation is where one can define to the model that there is censoring (of any type), see chapter 2 [here](https://cran.r-project.org/web/packages/flexsurv/vignettes/flexsurv.pdf). 

However, you can have a black box produce a likelihood (just make sure it output a non-negative value), use that in the censoring likelihood equation, and maximize this likelihood equation . That's kinda what I did with my experimental project [lifelike](https://github.com/CamDavidsonPilon/lifelike).
